tests/integration/test_automation/testParser.py

```diff
@@ -0,0 +1,507 @@
+           if search_address < test_common_yaml['memory']['regions'][test_yaml['test']['memory_region'] + '_beginning'] or search_address > test_common_yaml['memory']['regions'][test_yaml['test']['memory_region'] + '_end']:    
+               print( Fore.RED + 'The address ' + str(hex(search_address)) + ' is outsided the targetted memory region')
+               print(Fore.RED + str(hex(test_common_yaml['memory']['regions'][test_yaml['test']['memory_region'] + '_beginning'])))
+               print(Fore.RED + str(hex(test_common_yaml['memory']['regions'][test_yaml['test']['memory_region'] + '_end'])))
+               early_exit = True
+               break
+
+
+           #for each message in the  group, check if is in the log and has the expected parameters
+           for j in range(0,len( test_yaml['test']['groups'][i]['messages'])):
+
+               log_file.write("\t" +test_yaml['test']['groups'][i]['messages'][j]['log'] + "\t")
+               message_found = False
+               definite_message = True
+               #if the message is tagged as uncertain, tell the checker. This type of message is difficult to predict. i.e. writeback evicts
+               if "not_certain" in test_yaml['test']['groups'][i]['messages'][j]:
+                   definite_message = False
+               message_found = list_checker_dict[ test_yaml['test']['groups'][i]['messages'][j]['log']].checker(i,j, definite_message) 
+               # if message with parameters has not been found, output log data, and do an early exit                               
+               if message_found == False:
+                   print(Fore.RED + "Expected  message " + test_yaml['test']['groups'][i]['messages'][j]['log'] + " " +str(j) + " of group " + str(i) + " not found")
+                   print(Fore.RED + "at address " + str(hex(search_address)))
+                   for i in range(0, len(test_yaml['test']['groups'])):
+                      print("next group " + str(i))
+                      for k in range(0, len(test_yaml['test']['groups'][i]['messages'])):
+                         print(test_yaml['test']['groups'][i]['messages'][k])
+                   early_exit = True
+                   break              
+
+
+   #if there is extra messages not expected by the test sequence, flag this as fail.                 
+   extra_messages = are_there_any_extra_messages()
+
+   if early_exit == True or extra_messages == True:
+      print(Fore.RED + "Test FAILED !! \n")
+      log_file.write("Test FAILED !! \n")
+      if "test_notes" in test_yaml['test']:
+           print(Fore.YELLOW + test_yaml['test']['test_notes'])
+           log_file.write(test_yaml['test']['test_notes'])           
+   else:
+      print(Fore.GREEN + "Test PASSED !! \n ")
+      log_file.write("Test PASSED !! \n")
+
+   print_warnings() 
+
+
+   log_file.close()
+   file1.close()
+   file2.close()
+   file.close()
+   print(Style.RESET_ALL)
+
+
```

Comment:
I would add a return value, which can be used by e.g. the Makefile or other scripts to automate the workflow

Reply:
Done albeit with a sys.exit. This is because you would need to encapsulate all of the code under main into a function , and then call it from main, as to attach a return value. That becomes problematic because that code has some globals that other functions refer to.

Reply:
OK